src/ASM/ASMs2D.C

```diff
@@ -2651,10 +2653,12 @@ void ASMs2D::generateThreadGroups (size_t strip1, size_t strip2, bool silence)
 #endif
       // Verify that the nodes on this thread are not present on the others
       for (k = 0; k < j; k++)
-        for (nit = nodes[j].begin(); nit != nodes[j].end(); ++nit)
-          if (nodes[k].find(*nit) != nodes[k].end())
+        for (nit = nodes[j].begin(); nit != nodes[j].end(); ++nit) {
+          if ((!this->isLMn(*nit+1) || this->getLMType(*nit+1) != 'G' || !ignoreGlobalLM) &&
+               nodes[k].find(*nit) != nodes[k].end())
```

Comment:
Jeg bare lurer, er det tilfeller der man faktisk _vil_ ha sanity-sjekken for de globale LM'ene? Vil ikke de alltid være OK? Da kunne man slippe å propagere ignoreGlobalLM gjennom kallene. Dessuten er det overflødig å sjekke både isLMn og getLMType != 'G'. Hvis LMType er 'G' vil også isLMn alltid være true, faktisk gjøres isLMn inni getLMType også. Og endelig, ikke nødvendig å innføre braces her da det bare er ett statement i loopen.